### PR TITLE
Added documentation to describe how to use with AAD Auth Proxy

### PR DESCRIPTION
**Pull request by codesmon**
**Description:** Added documentation to describe how to use the AAD Auth Proxy to enable AAD based authentication.

**Link to tracking Issue:** 24451

**Testing:** None - documentation change.

**Documentation:** New AUTHENTICATION.md page describing how to configure the Exporter and the Auth Proxy to work together.

### CODE REVIEW

**Review by pcwiese — APPROVED** [review]

File: exporter/azuremonitorexporter/AUTHENTICATION.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+```yaml
+exporters:
+   azuremonitor:
+      instrumentation_key: ${env:APP_INSIGHTS_KEY}
+```
+
+## AAD/Entra Authentication
+
+Local Authentication can be disabled in [Applicaiton Insights](https://learn.microsoft.com/en-us/azure/azure-monitor/app/azure-ad-authentication) and an AAD based identity can be used in conjunciton with the instrumentation key. The instrumentation key is still required by the Exporter, but it is no long  sufficient to just have this in order to import data.
```
[the remark this answers is not in the document]
Comment (codesmon) [reply exporter/azuremonitorexporter/AUTHENTICATION.md]: Fixed a couple of typos :)

File: exporter/azuremonitorexporter/AUTHENTICATION.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ exporters:
 
 ## AAD/Entra Authentication
 
-Local Authentication can be disabled in [Applicaiton Insights](https://learn.microsoft.com/en-us/azure/azure-monitor/app/azure-ad-authentication) and an AAD based identity can be used in conjunciton with the instrumentation key. The instrumentation key is still required by the Exporter, but it is no long  sufficient to just have this in order to import data.
+Local Authentication can be disabled in [Applitiiton Insights](https://learn.microsoft.com/en-us/azure/azure-monitor/app/azure-ad-authentication) and an AAD based identity can be used in conjunction with the instrumentation key. The instrumentation key is still required by the Exporter, but it is no long  sufficient to just have this in order to import data.
```
Comment (jpkrohling) [inline exporter/azuremonitorexporter/AUTHENTICATION.md]:
```suggestion
Local Authentication can be disabled in [Application Insights](https://learn.microsoft.com/en-us/azure/azure-monitor/app/azure-ad-authentication) and an AAD-based identity can be used in conjunction with the instrumentation key. The instrumentation key is still required by the Exporter, but it is no longer sufficient to just have this to import data.
```

**Review by jpkrohling — APPROVED** [review]
I didn't try the new instructions, but if code owners are happy, this can be merged.

**Review by dashpole — APPROVED** [review]
Also didn't try out the instructions, but lgtm

**Pull request merged**